spaghetti/network.py

```diff
@@ -156,13 +158,13 @@ def _extractnetwork(self):
                 v = self._round_sig(v)
                 try:
                     vid = self.nodes[v]
-                except:
+                except BaseException:
```

Comment:
I think this should be a `KeyError`, but I'm guessing here. It looks like the `try` sets variable `vid` to the value stored in the cache, `self.nodes`, under key `v`. 
But, if the cache does not have `v`, then it sets both `self.nodes[v]` and `vid` to `nodecount`. 

The only way I see this generating an exception is if `nodes[v]` doesn't have `v`, which would be a `KeyError` under most idioms.

Reply:
This was part of some code that we put in several years back, and I think you are right. I would hazard to guess that we put it in for handling #9

Reply:
yeah, so definitely don't take out the `except` overall, only change the exception type to match the exception we're interested in. 

This `try/except` is the right solution to that problem but we just need to make sure the type of exceptions are right.

Reply:
same as [above](https://github.com/pysal/spaghetti/pull/60#pullrequestreview-144623191). It's just a try/except. Do we need to use `KeyError`?

Reply:
sorry, I'm not being as explicit as I can be: 

The "bare" try/except (which your pep-8 fixer is complaining about and trying to fix) will match *any* error thrown inside of it. However, if you specify the *type* of error you expect to be thrown, only that error will get caught. So, here, 
```python
try:
    vid = self.nodes[v]
except KeyError:
    self.nodes[v] = vid = nodecount 
```
this will only execute `self.nodes[v] = vid = nodecount` if a `KeyError` (or a subclass of a `KeyError` is raised. 

So, say like... if `self` has no attribute `nodes`, the statement in the `try` block would raise an `AttributeError`, not a `KeyError`, since `self` has no attribute `nodes`, and so can not fail to find `v` in its keys. Instead, if you do:
```python
try:
    vid = self.nodes[v]
except:
    self.nodes[v] = vid = nodecount 
```
or what it was changed to:
```python
try:
    vid = self.nodes[v]
except BaseException:
    self.nodes[v] = vid = nodecount 
```
any exception inside of the `try` will get "caught," and the `except` clause executed, regardless if that's the exception we are trying to catch. 

Does that make sense?